content/news/2019-12-03-mapstruct-and-quarkus.md

```diff
@@ -0,0 +1,220 @@
+
+Read [this article](http://in.relation.to/2019/03/08/why-quarkus/) if you want to know more about Quarkus itself.
+
+#### And what is so special when using it with MapStruct?
+The special thing about Quarkus is that it not only allows to run applications on the JVM, but also as native binaries via [GraalVM](https://www.graalvm.org/). This provides ahead-of-time compilation that creates a native image of your application (native system-dependent machine code and not bytecode). Thus you don’t need to have an own JVM installation to run the code, similar to a compiled C++ or Go application. \
+This allows having the advantages of native applications also for Java-based apps like a much faster startup time, less memory usage and a much smaller image (just a few megabytes instead of easily more than 100 MB for the JVM and required libraries).
+
+Using a native image is not possible for all kinds of applications, especially when reflection is used the GraalVM compiler needs some assistance by the developer in order to create the native image. That means you have to provide reflection metadata to the compiler which makes the development more complicated.
+
+And thats exactly the point where MapStruct jumps in as a reflection-free mapping library!
+
+### How to integrate MapStruct in a Quarkus application
+So let us see how well MapStruct plays together with Quarkus!
+
+I made an example application that will show you how you can integrate MapStruct in your Quarkus application. You will find the complete demo application in our [examples repository](https://github.com/mapstruct/mapstruct-examples/mapstruct-quarkus).
+
+The example should be a simple REST application just having a `/person` endpoint returning information about a person. Obviously, the internally used classes should not be exposed, but a DTO will be created and filled with information from a MapStruct mapper.
+
+#### Create a new Quarkus application
+After reading the [Getting Started](https://quarkus.io/guides/getting-started-guide) on the Quarkus pages it was really easy to figure out how to create a new application. Just run the following command:
+
+{{< prettify >}}
+mvn io.quarkus:quarkus-maven-plugin:1.0.1.Final:create \
+  -DprojectGroupId=org.mapstruct.examples.quarkus \
+  -DprojectArtifactId=mapstruct-examples-quarkus \
+  -DclassName="org.mapstruct.example.quarkus.PersonResource" \
+  -Dpath="/person" \
+  -Dextensions="resteasy-jsonb"
+{{< /prettify >}}
+
+This will setup a new Quarkus application having a REST endpoint `/person` defined in a `PersonResource`. We want to expose the JSON structure automatically, so we need the `resteasy-jsonb` extension that will allow us to marshall our object to JSON (like you probably know it from Spring REST or similar frameworks).
```

Comment:
Might be useful to mention we also support Jackson. People are usually more familiar with it.

(just a suggestion, might be a bit too much as it's not the purpose of this blog post)